client/jetpack-connect/test/utils.js

```diff
@@ -60,3 +65,27 @@ describe( 'getRoleFromScope', () => {
 		expect( result ).toBe( null );
 	} );
 } );
+
+describe( 'parseAuthorizationQuery', () => {
+	test( 'should return transformed data on valid input', () => {
+		const data = {
+			_wp_nonce: 'foobar',
+			blogname: 'Just Another WordPress.com Site',
+			client_id: '12345',
+			home_url: 'http://yourjetpack.blog',
+			redirect_uri: 'http://yourjetpack.blog/wp-admin/admin.php',
+			scope: 'administrator:34579bf2a3185a47d1b31aab30125d',
+			secret: '640fdbd69f96a8ca9e61',
+			site: 'http://yourjetpack.blog',
+			site_url: 'http://yourjetpack.blog',
+			state: '1',
+		};
+		const result = parseAuthorizationQuery( data );
+		expect( result ).not.toBeNull();
```

Comment:
This assertion seems unnecessary IMHO, considering the snapshot contains an object.

Reply:
The snapshot _could be_ of null. Snapshot tests are, in some ways, opaque. Especially when you first add them. I believe there's value in documenting explicit expectations in addition to snapshots.